### PR TITLE
[PROBABLY SHITCODE] Inspection Prefs!

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request

Both easier and harder than I expected, for a few reasons.

For now, shamelessly steals some UI layout inspiration from SR, but otherwise, all original.

## How Does This Help ***Gameplay***?

Being able to write text dumps about your character is neat!

Also having parts of it hide depending on what can be seen is also incredibly nice! No having to put all your visual information inside OOC text, if you were a mask enjoyer on SR.

OOC text is not implemented, because I'm going to be integrating it into the matchmaking system, so I'd have to rewrite that section anyways.

## How Does This Help ***Roleplay***?

Less second guessing on who you're RPing with, very nice!

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary> 

image
image
image
image
image
image

</details>

## Changelog

:cl:
add: Inspection prefs, check them out! Parts of it hide depending on your outfit!
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/inspection.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  name: 'Broad Inspection Text',
+  description:
+    "A longer text description of your character's overall hard to conceal attributes, like height.",
+  component: FeatureTextInput,
+};
+
+export const face_inspection_text: Feature<string> = {
+  name: 'Face Inspection Text',
+  description:
+    "A longer text description of your character's face. Hidden when your eyes/face is obscured.",
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/inspection.tsx:19]: Good catch

File: code/modules/client/client_defines.dm [hunk 2]
```diff
@@ -267,3 +267,6 @@
 	/// Holder variables for viewport scaling.
 	var/last_view_x_dim = 7
 	var/last_view_y_dim = 7
+
+	/// Whether the client is viewing OOC information in the inspect menu or not.
+	var/viewing_ooc_info = TRUE
```
Comment (nevimer) [inline code/modules/client/client_defines.dm:272]: on the client? not the panel itself? what if I am looking at multiple OOC informations at once?

File: code/modules/mob/living/carbon/human/inspection_panel.dm [hunk 3]
```diff
@@ -0,0 +1,39 @@
+/datum/inspection_panel
+	/// Mob that the examine panel belongs to.
+	var/mob/living/holder
+
+/datum/inspection_panel/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/inspection_panel/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
```
Comment (nevimer) [inline code/modules/mob/living/carbon/human/inspection_panel.dm:9]: I feel like the two mobs should be able to see each other before this updates.

File: code/__HELPERS/text.dm [hunk 4]
```diff
@@ -1188,3 +1188,13 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 /proc/sanitize_css_class_name(name)
 	var/static/regex/regex = new(@"[^a-zA-Z0-9]","g")
 	return replacetext(name, regex, "")
+
+/// Returns a or an depending on what the text starts with.
+/proc/prefix_a_or_an(text)
```
Comment (nevimer) [inline code/__HELPERS/text.dm:1193]: I feel like this is better described as a vowel check, so someone doesn't write one again. 

/proc/piglatin_word(word) - Exists already and basically does this, for example, so yeah, just thinking of duplicate code in the future showing up when it can be kinda unified for future use purposes.

**Review by nevimer — COMMENTED** [review]
Just some feedback.

File: code/modules/client/client_defines.dm (hunk 2 above)
Comment (RimiNosha) [reply code/modules/client/client_defines.dm:272]: Then they'll both show OOC info. There's very little point in adding a state per panel, cause you're pretty unlikely to inspect the same person every time, and that's a bunch of times you'll be finding yourself hiding OOC info.

File: code/modules/mob/living/carbon/human/inspection_panel.dm (hunk 3 above)
Comment (RimiNosha) [reply code/modules/mob/living/carbon/human/inspection_panel.dm:9]: I decided to let the game keep updating the panel, due to the fact it's not showing a literal live preview of their being, unlike SR.

File: code/__HELPERS/text.dm (hunk 4 above)
Comment (RimiNosha) [reply code/__HELPERS/text.dm:1193]: `piglatin_word` does not do this at all, that is for applying a sol accent to a sentence spoken ICly. Not that it's actually ever used AFAIK.

`a_or_an` makes the most sense, cause it's in plain english, and is pretty to the point.

**Comment by RimiNosha** [comment]
image
